pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/WorkerImpl.java

```diff
@@ -131,33 +125,45 @@ public Map<String, Collection<String>> getAssignments(String clientRole) {
         return ret;
     }
 
-    private boolean isSuperUser(final String clientRole) {
-        return clientRole != null && worker().getWorkerConfig().getSuperUserRoles().contains(clientRole);
+    private void throwIfNotSuperUser(Authentication authentication, String action) {
+        if (authentication.getClientRole() != null) {
```

Comment:
this looks risky

Reply:
Fantastic catch. I implemented this incorrectly. Thank you!